docs/How_To/Tutorials/document_reader.rst

```diff
@@ -0,0 +1,253 @@
+
+    agent_config: # For the demo, default configuration is used. You can change the configuration as per your requirement
+        _target_: sherpa_ai.config.task_config.AgentConfig
+
+
+    llm:  # Configuration for the llm, here we are using the OpenAI GPT-3.5-turbo model
+        _target_: langchain.chat_models.ChatOpenAI
+        model_name: gpt-3.5-turbo
+        temperature: 0.7
+
+    embedding_func: 
+        _target_: langchain.embeddings.SentenceTransformerEmbeddings
+        model_name: sentence-transformers/all-mpnet-base-v2
+
+    doc_search:
+        _target_: actions.DocumentSearch
+        filename: paper.pdf
+        embedding_function: ${embedding_func}
+        k: 4
+
+    qa_agent:
+        _target_: sherpa_ai.agents.qa_agent.QAAgent
+        llm: ${llm}
+        shared_memory: ${shared_memory}
+        name: QA Sherpa
+        description: You are a Question answering assistant helping users to find answers to the text. Based on the input question, you will provide the answer from the text ONLY.
+        agent_config: ${agent_config}
+        num_runs: 1
+        actions:
+            - ${doc_search}
+
+
+The `_target_` key is used to define the class to be used when instantiating the object. 
+
+The DocumentSearch action is defined in the `doc_search` field, you can change the `filename` attribute to the PDF file you want to use. The `embedding_function` attribute is set to the SentenceTransformerEmbeddings class, which is used to convert text into vectors. The `k` attribute is set to 4, which is the number of search results to return. 
+
+The agent is defined in the last section of this configuration file called `qa_agent.` It has the following parts:
+
+1. `llm`: This is the language model used by the agent. In this case, we are using the OpenAI GPT-3.5-turbo model.
+
+2. `shared_memory`: This is the shared memory used by the agent. The shared memory is used to store information that can be shared between different agents. Since we only have one agent in this tutorial, we can choose the default shared memory. There will be a separate tutorial on how to use shared memory.
+
+3. `name` and `description`: These are used to describe the agent when it is executing the task.
+
+4. `agent_config`: This is the configuration for the agent. The default configuration is used in this tutorial.
+
+5. `num_runs`: This is the number of runs the agent will execute an action. In this tutorial, the agent will execute only once.
+
+6. `actions`: This is the list of actions that the agent can execute. In this case, the agent can execute the `doc_search` action.
+
+
+Put it all together
+********************
+
+Now, we can put everything together to create the PDF reader. Create a file called `main.py` and add the following code:
+
+.. code-block:: python
+
+    from argparse import ArgumentParser
+
+    from hydra.utils import instantiate
```

Comment:
do I need to pip install these? hydra? argparse? etc

Reply:
No, these are install as part of the sherpa_ai installation